### PR TITLE
feat: adds profile switcher

### PR DESCRIPTION
**Pull request by zomars**
## What does this PR do?

Adds a basic profile switcher. Allows multi-account users to switch between them easily.

image

## How does it work?

On the database, you create a User<->User relationship. This allows them to switch between those profiles easily. If there's demand for this feature we could come up with a way to self-serve profile linking. Out of scope for this PR.

https://github.com/calcom/cal.com/assets/3504472/056047ab-23cd-4276-a27d-46fa8087ba42

## Is it safe?

Right now the only way to link an account is manually via DB access. So there's no entry point for abuse or vulnerabilities.

## Why?

At first I've made this as a way to switch easily between my testing accounts locally. But I'm contributing this feature since it may come handy for admins or users that have multiple accounts (org, teams, personal, etc.)

## Requirement/Documentation

- If there is a requirement document, please, share it here.
- If there is ab UI/UX design document, please, share it here.

## Type of change

- New feature (non-breaking change which adds functionality)

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Mandatory Tasks

- [ ] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by zomars** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #13408** <a href="https://app.graphite.dev/github/pr/calcom/cal.com/13408?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/calcom/cal.com/13408?utm_source=stack-comment).

File: packages/features/shell/Shell.tsx [hunk 1]
```diff
@@ -898,7 +900,7 @@ function SideBar({ bannersHeight, user }: SideBarProps) {
         <div className="flex h-full flex-col justify-between py-3 lg:pt-4">
           <header className="todesktop:-mt-3 todesktop:flex-col-reverse todesktop:[-webkit-app-region:drag] items-center justify-between md:hidden lg:flex">
             {orgBranding ? (
-              <Link href="/settings/organizations/profile" className="w-full px-1.5">
+              <Link href="/settings/organizations/profile" className="px-1.5">
```
Comment (zomars) [inline packages/features/shell/Shell.tsx:903]: Before

image

After

image

**Review by zomars — COMMENTED** [review]
Self review done

**Comment by leog** [comment]
@zomars just leaving my previous attempt for this feature here in case it helps in some way: https://github.com/calcom/cal.com/pull/11627

**Comment by zomars** [comment]
> @zomars just leaving my previous attempt for this feature here in case it helps in some way: https://github.com/calcom/cal.com/pull/11627

Thank you @leog Although this approach doesn't require the user to have the same emails. Aimed for simplicity even if this means manual linking from an admin on the first iteration.

**Comment by hariombalhara** [comment]
I am not sure how the two Profile switchers would work together.
1. A Profile Switcher that switches b/w different Users that are linked together to be owned by same user but have different emails - Better term for this could be **User Switcher**
2. A Profile switcher that allows a user to switch b/w different profiles that are part of different organizations but have the same email. This is something [that I am almost done with here](https://github.com/calcom/cal.com/pull/13002). It was a requirement to be able to join and leave an organization.

**Comment by zomars** [comment]
> I am not sure how the two Profile switchers would work together.
> 
> 1. A Profile Switcher that switches b/w different Users that are linked together to be owned by same user but have different emails - Better term for this could be **User Switcher**
> 2. A Profile switcher that allows a user to switch b/w different profiles that are part of different organizations but have the same email. This is something [that I am almost done with here](https://github.com/calcom/cal.com/pull/13002). It was a requirement to be able to join and leave an organization.

Agreed @hariombalhara I wasn't aware about your profile switcher. Agreed that User switcher is a more appropriate name. Kinda like Google lets you logging to multiple accounts at the same time although the it is not quite the same implementation. In the current state is more like a debugging tool or commodity.

**Comment by keithwillcode** [comment]
Moving back to draft while merge conflicts are resolved and we figure out how we want to proceed with this.

**Comment by PeerRich** [comment]
with the new mono-account approach of @hariombalhara i dont think we need a profile switcher

others dont really have multiple accounts either, and if they have two accounts, its quiet rare they want to link them. is it ok if i close this @zomars @keithwillcode @hariombalhara ?

**Pull request closed without merging**